consensus/core/src/core.rs

```diff
@@ -669,40 +737,48 @@ impl Core {
             }
             assert!(commits_until_update > 0);
 
-            // TODO: limit commits by commits_until_update, which may be needed when leader schedule length
-            // is reduced.
-            let decided_leaders = self.committer.try_decide(self.last_decided_leader);
+            // Always try to process the synced commits first
+            let mut decided_leaders = self
+                .committer
+                .try_decide_synced(&mut synced_commits, commits_until_update);
+
+            // If the synced `decided_leaders` is empty then try to run the decision rule.
+            if decided_leaders.is_empty() {
```

Comment:
Why can't you update self.last_decided_leader from the synced decided leaders and then run the commit rule to fill whatever remaining space is left for the schedule?